### PR TITLE
PEM to JWK converters

### LINKED ISSUE #4321: Add a PEM public key to JWK converter in the JS API
For both elliptic curves and RSA keys,

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves #4321 

This PR adds C++ and JS/TS API to convert a public/private EC/RSA key from PEM to JSON Web Key (JWK). Note that this uses the Python [`JWCrypto`](https://pypi.org/project/jwcrypto/) package to compare the results to a well-known implementation.

TODO: 
- [x] TS polyfill

### CODE REVIEW

File: include/ccf/crypto/jwk.h [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+namespace crypto
+{
+  enum class JsonWebKeyType
+  {
+    EC = 0,
+    RSA = 1
+  };
+  DECLARE_JSON_ENUM(
+    JsonWebKeyType, {{JsonWebKeyType::EC, "EC"}, {JsonWebKeyType::RSA, "RSA"}});
```
Comment (achamayou) [inline include/ccf/crypto/jwk.h:19]: Not urgent, but I think some users will want ED/Curve25519 support eventually,

File: include/ccf/crypto/jwk.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  {
+    switch (curve_id)
+    {
+      case CurveID::SECP384R1:
+        return JsonWebKeyECCurve::P384;
+      case CurveID::SECP256R1:
+        return JsonWebKeyECCurve::P256;
+      case CurveID::SECP256K1:
+        return JsonWebKeyECCurve::P256K1;
+      default:
```
Comment (wintersteiger) [inline include/ccf/crypto/jwk.h:58]: +P521?

File: tests/js-modules/modules.py [hunk 3]
```diff
@@ -494,37 +551,23 @@ def test_npm_app(network, args):
...
-            {
-                "algorithm": algorithm,
-                "key": key_pub_pem,
-                "signature": b64encode(signature).decode(),
-                "data": b64encode(data).decode(),
-            },
-        )
-        assert r.status_code == http.HTTPStatus.OK, r.status_code
-        assert r.body.json() == True, r.body
+        curves = [ec.SECP256R1, ec.SECP256K1]
```
Comment (wintersteiger) [inline tests/js-modules/modules.py:554]: With 384/512 we would use different hash algorithms, not worth adding to this test though I think.

**Review by wintersteiger — APPROVED** [review]
LGTM! Somebody already said this somewhere and I agree, Curve25519 would be a worthwhile addition.

File: include/ccf/crypto/jwk.h [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+#include "ccf/ds/logger.h"
+
+#include <string>
+
+namespace crypto
+{
+  enum class JsonWebKeyType
+  {
+    EC = 0,
+    RSA = 1
```
[the remark this answers is not in the document]
Comment (jumaffre) [reply include/ccf/crypto/jwk.h:16]: Yes. I haven't included this in this PR as this is only support for asymmetric keys for now.

File: include/ccf/crypto/jwk.h (hunk 2 above)
Comment (jumaffre) [reply include/ccf/crypto/jwk.h:59]: This isn't supported yet (see `CurveID` enum) so I've omitted it from this PR.

File: tests/js-modules/modules.py [hunk 5]
```diff
@@ -37,6 +39,61 @@ def validate_openapi(client):
         raise e
 
 
+def generate_and_verify_jwk(client):
+    LOG.info("Generate JWK from raw public key PEM")
+    r = client.post("/app/pubPemToJwk", body={"pem": "invalid_pem"})
+    assert r.status_code != http.HTTPStatus.OK
+
+    # Elliptic curve
+    curves = [ec.SECP256R1, ec.SECP256K1, ec.SECP384R1]
```
[the remark this answers is not in the document]
Comment (jumaffre) [reply tests/js-modules/modules.py:48]: Same as above.

File: include/ccf/crypto/jwk.h [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  };
+  DECLARE_JSON_ENUM(
+    JsonWebKeyType, {{JsonWebKeyType::EC, "EC"}, {JsonWebKeyType::RSA, "RSA"}});
+
+  struct JsonWebKeyBase
```
[the remark this answers is not in the document]
Comment (jumaffre) [reply include/ccf/crypto/jwk.h]: Good point. This is now done.

**Review by achamayou — APPROVED** [review]

**Pull request merged**